### PR TITLE
DOC: Add note in 1.5.3 about SQLAlchemy warnings

### PR DESCRIPTION
**Pull request by mroeschke**
xref #50558 

As discussed in todays dev call, for 1.5.3 the decision was to document that SQLAlchemy warnings can be expected but not addressed.

### CODE REVIEW

File: doc/source/whatsnew/v1.5.3.rst [hunk 1]
```diff
@@ -37,6 +37,12 @@ Bug fixes
 
 Other
 ~~~~~
+
+.. note::
+
+    If you are using :meth:`DataFrame.to_sql`, :func:`read_sql`, :func:`read_sql_table`, or :func:`read_sql_query` with SQLAlchemy 1.4.6 or greater,
```
Comment (phofl) [inline doc/source/whatsnew/v1.5.3.rst:43]: 1.4.46

File: doc/source/whatsnew/v1.5.3.rst [hunk 2]
```diff
@@ -37,6 +37,12 @@ Bug fixes
...
+    you may see a ``sqlalchemy.exc.RemovedIn20Warning``. These warnings can be safely ignored as pandas works toward compatibility with SQLAlchemy 2.0.
```
Comment (phofl) [inline doc/source/whatsnew/v1.5.3.rst:44]: Can we reword the last part? These warnings can be safely ignored for the SQLAlchemy 1.4.x releases as pandas works towards compatibility ...

File: doc/source/whatsnew/v1.5.3.rst (hunk 1 above)
Comment (mroeschke) [reply doc/source/whatsnew/v1.5.3.rst]: Thanks fixed

File: doc/source/whatsnew/v1.5.3.rst (hunk 2 above)
Comment (mroeschke) [reply doc/source/whatsnew/v1.5.3.rst]: Good idea. Changed

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @mroeschke

**Comment by phofl** [comment]
@meeseeksmachine backport 1.5.x

**Comment by phofl** [comment]
@meeseeksdev backport to 1.5.x

**Pull request merged**